src/openapi/streaming/connection/transport/websocket-transport.js

```diff
@@ -181,58 +189,34 @@ function handleSocketMessage(messageEvent) {
 }
 
 function handleSocketClose(event) {
-    if (
-        (this.socket && this.socket.readyState === STATE_INTERNAL_CLOSED) ||
-        this.socket.readyState === STATE_INTERNAL_CLOSING
-    ) {
-        log.debug(LOG_AREA, 'Transport connection closed', {
-            readyState: this.socket.readyState,
-            code: event.code,
-        });
-
-        // 1006 doesn't tell us its a 401 unauthorized, just that it might be
-        // the websocket spec disallows reading status codes for security reasons
-        // (so websocket can't be used to probe http endpoints)
-        // So, before we reconnect, lets re-authorize to be sure
-        // and if that gets a 401, that will report the token invalid via
-        // authTransport
-        if (event.code === 1006) {
-            this.getAuthorizePromise(this.contextId, true);
-        }
-
-        // Order here is important. We need to invoke state callback first, as parent streaming manager updates query upon disconnect.
-        this.stateChangedCallback(constants.CONNECTION_STATE_DISCONNECTED);
-        reconnect.call(this);
+    if (!this.socket) {

```

Comment:
In which scenario is it possible to get empty this.socket state? I don't think it's possible or should be. If we plan to clean the this.socket instance, we should also remove callbacks before hand.

Reply:
Is it possible to happen in current disconnect method where we are cleanly disconnecting?

Reply:
nope... unless we call this method by mistake
it has reconnection logic and it could lead to hair pulling bugs
Earlier it was guarded too and I thought it was for same reason ?